lib/matplotlib/artist.py

```diff
@@ -43,30 +42,23 @@ def allow_rasterization(draw):
     other setup function calls, such as starting and flushing a mixed-mode
     renderer.
     """
-    @contextmanager
-    def with_rasterized(artist, renderer):
-
-        if artist.get_rasterized():
-            renderer.start_rasterizing()
-
-        if artist.get_agg_filter() is not None:
-            renderer.start_filter()
 
+    # the axes class has a second argument inframe for its draw method.
+    @wraps(draw)
```

Comment:
Do not have is an issue with this, but why?

Reply:
Because I don't think adding an extra layer of indirection really helps with clarity -- does it?

Reply:
No, think this will make many tracebacks a few levels shorter.  I asked this before I fully groked that this is all inside of a decorator!